api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -619,34 +648,41 @@ type ImageContentSource struct {
 	Mirrors []string `json:"mirrors,omitempty"`
 }
 
-// ServicePublishingStrategyMapping specifies how individual control plane
-// services are published from the hosting cluster of a control plane.
+// ServicePublishingStrategyMapping specifies how individual control plane services endpoints are published for consumption.
+// This includes APIServer;OAuthServer;Konnectivity;Ignition.
+// If a given service is not present in this list, it will be exposed publicly by default.
 type ServicePublishingStrategyMapping struct {
-	// Service identifies the type of service being published.
+	// service identifies the type of service being published.
+	// It can be APIServer;OAuthServer;Konnectivity;Ignition
+	// OVNSbDb;OIDC are no-op an kept for backward compatibility.
+	// This field is immutable.
 	//
 	// +kubebuilder:validation:Enum=APIServer;OAuthServer;OIDC;Konnectivity;Ignition;OVNSbDb
 	// +immutable
 	Service ServiceType `json:"service"`
 
-	// ServicePublishingStrategy specifies how to publish Service.
+	// servicePublishingStrategy specifies how to publish a service endpoint.
 	ServicePublishingStrategy `json:"servicePublishingStrategy"`
 }
 
 // ServicePublishingStrategy specfies how to publish a ServiceType.
+// +kubebuilder:validation:XValidation:rule="(self.type == 'NodePort' && has(self.nodePort) && !has(self.loadBalancer) && !has(self.route)) || (self.type == 'LoadBalancer' && !has(self.nodePort) && !has(self.route)) || (self.type == 'Route' && !has(self.nodePort) && !has(self.loadBalancer)) || ((self.type == 'None' || self.type == 'S3') && !has(self.nodePort) && !has(self.loadBalancer) && !has(self.route))",message="Only the field corresponding to the 'type' should be set, or none if Type is 'None' or 'S3'."
```

Comment:
This is easier when you break it down into multiple ternarys
```
// +kubebuilder:validation:XValidation:rule="self.type == 'NodePort' ? has(self.nodePort) : !has(self.nodePort)",message="nodePort is required when type is NodePort, and forbidden otherwise"
```